dbms/src/Common/setThreadName.cpp

```diff
@@ -29,11 +29,10 @@ void setThreadName(const char * name)
         throw DB::Exception("Thread name cannot be longer than 15 bytes", DB::ErrorCodes::PTHREAD_ERROR);
 #endif
 
-#if defined(__FreeBSD__)
+#if defined(OS_FREEBSD)
     pthread_set_name_np(pthread_self(), name);
-    return;
-
-#elif defined(__APPLE__)
+    if ((false))
```

Comment:
Looks strange.

Reply:
Clang issues a warning with this specific workaround for explicitly dead code.